general/bundler_imports.md

```diff
@@ -0,0 +1,160 @@
+
+3. **No version constraints in `package.json`, only in `.npmbundlerrc`**: this is a degenerate case of 2 where you simply omit `react` in your build. The benefits are that, if you don't need it for the build, you don't download or care about it. However, it may be considered wrong because you are not declaring one of your dependencies in the `package.json` file even though it will be used during runtime.
+
+> You can see all the implicit imports you get when building a `DXP` module in the [default npm bundler preset](https://github.com/liferay/liferay-npm-tools/blob/master/packages/liferay-npm-bundler-preset-liferay-dev/config.json), that gets injected by `liferay-npm-scripts` when you run `gradle build`. Look for `"imports"` inside the `"config"` section.
+
+> Note that strategy 2 has to be carefully maintained because if you specify a very different version in `package.json` and `.npmbundlerrc` and develop your project targeting the version in `package.json`, you may have problems during runtime if your provider's version is not compatible. This is not the case in `DXP` where we coordinate together to align JavaScript version, but it is worth mentioning the risk.
+
+## How npm bundler imports works at runtime
+
+The technique used to make imports work is quite simple.
+
+Say you import `react` from `frontend-js-react-web` in your `A` project, then every appearance of `react` in:
+
+1. Your project's source code
+2. Your project's dependencies (packages in `node_modules`)
+
+will be replaced by `frontend-js-react-web$react` instead of `A$react` as would be the default. After that, the resulting `package.json` will have a dependency to `frontend-js-react-web$react` injected with the version constraints specified in the `.npmbundlerrc` import configuration so that it can be resolved by the AMD loader at runtime.
+
+This way, code in your project like the following:
+
+```javascript
+import React from 'react';
+```
+
+will bind the `React` variable the `frontend-js-react-web`'s copy of `react` inside the browser's JavaScript interpreter.
+
+Note that, once your code "jumps" into `frontend-js-react-web`'s copy of `react`, your project's configuration doesn't play any effect until your return from that "jump". This is because once you invoke `React` from`frontend-js-react-web$react` you enter the bundled JavaScript code of `frontend-js-react-web` which has been bundled by the settings in the `frontend-js-react-web` project, not yours.
```

Comment:
- "play any effect" → "have any effect".
- missing space in "from`frontend-js-react-web$react`.